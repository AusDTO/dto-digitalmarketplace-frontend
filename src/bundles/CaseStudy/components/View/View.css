main .sidebar {
  float: left;
  margin: 0 2.04556% 0 0;
}

.actions {
  text-align: right;
}

.actions a {
  display: inline-block;
}
.actions a:first-of-type {
  margin-right: 1em;
}

header {
  border-bottom: 1px solid #cecece;
  padding-bottom: 35px;
}

header h2 {
  margin-bottom: 0;
}
