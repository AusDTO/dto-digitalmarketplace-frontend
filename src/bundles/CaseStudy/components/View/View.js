import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import isEmpty from 'lodash/isEmpty';

import { newline } from '../../../../helpers';

import view from './View.css'

class View extends React.Component {
  state = { showConfirm: false }

  toggleConfirm(show = true) {
    this.setState({
      showConfirm: show
    }, () => this.state.showConfirm && this.refs.confirm.focus());
  }

  render() {
    const {
      title,
      opportunity,
      client,
      supplier_name,
      supplier_url = null,
      approach,
      timeframe,
      outcome,
      project_links,
      service,
      roles,
      meta,
      confirmButton = null,
      returnLink = null
    } = this.props;

    const { showConfirm } = this.state;

    return (
      <section id="casestudy__view" styleName="view.case-study-summary">
        {showConfirm && (
          <div ref="confirm" className="callout--warn" aria-labelledby="callout--success__heading" tabIndex="-1" role="alert">
              <p id="callout--success__heading">Are you sure you want to delete this case study?</p>
              <a href={meta.deleteLink} className="button">Delete this case study</a>
              <button className="button-secondary" onClick={this.toggleConfirm.bind(this, false)}>No, keep this case study</button>
          </div>
        )}
        <header className="row">
          <div className="col-xs-12">
            <h1 className="au-display-xl" tabIndex="-1">{title}</h1>
          </div>

          <div className="meta col-xs-12">
            <div className="row">
              <div className="col-xs-12 col-sm-7">
                <p>by {supplier_url ? <a href={supplier_url}>{supplier_name}</a> : supplier_name}</p>
              </div>
              {meta && (
                <div className="col-xs-12 col-sm-5 actions">
                  <a href={meta.editLink}>Edit case study</a>
                  <button className="button-secondary" onClick={this.toggleConfirm.bind(this)}>Delete</button>
                </div>
              )}
            </div>
          </div>
        </header>
        <div className="row">
          <div styleName="view.case-study-content">
            <div styleName="view.case-study-sidebar">
              <aside className="col-sm-3 col-xs-12">
                <h3 className="au-display-md">Client</h3>
                <p>{client}</p>

                <h3 className="au-display-md">Timeframe</h3>
                <p>{timeframe}</p>

                <h3 className="au-display-md">Area of expertise</h3>
                <p>{service}</p>

                <h3 className="au-display-md">Responsible for</h3>
                <p>{roles}</p>
              </aside>
            </div>
            <article role="main" className="col-sm-7 col-xs-12">
              <section>
                <h2 className="au-display-md">Challenge</h2>
                <p className="freetext">{newline(opportunity)}</p>
              </section>
              <section>
                <h2 className="au-display-md">Approach</h2>
                <p className="freetext">{newline(approach)}</p>
              </section>
              <section>
                <h2 className="au-display-md">Outcomes and benefits</h2>
                <ul>
                  {outcome && outcome.map((content, i) => <li key={i}>{content}</li>)}
                </ul>
              </section>
              {project_links && project_links.length > 0 && (
                <section styleName="view.project">
                  <h2 className="au-display-md">Project Links</h2>
                  <ul>
                    {project_links.map((item, i) => <li key={i}>
                        {typeof item == 'object' ?
                            <a className="project__links" href={item.url} rel="external" target="_blank">{isEmpty(item.title) ? item.url : item.title}</a>
                            :
                            <a className="project__links" href={item} rel="external" target="_blank">{item}</a>
                        }
                    </li>)}
                  </ul>
                </section>
              )}
              <div className="casestudy__actions">
                {confirmButton}
                {returnLink}
              </div>
            </article>
          </div>
        </div>
      </section>
    )
  }
}

View.propTypes = {
  title: PropTypes.string.isRequired,
  opportunity: PropTypes.string.isRequired,
  client: PropTypes.string.isRequired,
  approach: PropTypes.string.isRequired,
  timeframe: PropTypes.string.isRequired,
  outcome: PropTypes.arrayOf(PropTypes.string).isRequired,
  project_links: PropTypes.arrayOf(PropTypes.string),
  meta: PropTypes.objectOf(PropTypes.string),

  returnLink: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...state.casestudy,
    ...ownProps
  }
}

export default connect(mapStateToProps)(View)
