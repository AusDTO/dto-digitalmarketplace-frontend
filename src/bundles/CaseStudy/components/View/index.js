import View from './View'
import sampleState from './View.json'

export { sampleState }
export default View
