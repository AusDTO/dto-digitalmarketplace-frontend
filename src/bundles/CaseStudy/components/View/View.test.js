// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import View from './View';
import renderer from 'react-test-renderer';

import sampleState from './View.json';
import createStore from '../../redux/create';

test('VerticalList with required attributes', () => {
  const store = createStore(Object.assign({}, { _serverContext: {} },  sampleState));
  const component = renderer.create(
    <View store={store} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('VerticalList with required attributes and meta links', () => {
  let state = Object.assign({}, sampleState);
  state.casestudy.meta = {
    editLink: 'link/to/edit',
    deleteLink: 'link/to/delete'
  };

  const store = createStore(Object.assign({}, { _serverContext: {} },  state));
  const component = renderer.create(
    <View store={store} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
