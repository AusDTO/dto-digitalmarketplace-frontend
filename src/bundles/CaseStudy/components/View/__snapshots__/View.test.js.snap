exports[`test View with required attributes 1`] = `
<section
  id="casestudy__view">
  <header
    className="row">
    <div
      className="col-xs-12">
      <h1>
        My Sydney
      </h1>
    </div>
    <div
      className="meta col-xs-12">
      <div
        className="row">
        <div
          className="col-xs-12 col-sm-7">
          <p>
            Transport for NSW
          </p>
        </div>
        <div
          className="col-xs-12 col-sm-5 actions">
          <a
            href="#">
            Edit case study
          </a>
          <button
            className="button-secondary"
            onClick={[Function]}>
            Delete
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    className="row">
    <aside
      className="col-sm-3 col-xs-12">
      <h4>
        Timeframe
      </h4>
      <p>
        January 2016 — June 2016
      </p>
      <h4>
        Service
      </h4>
      <p />
      <h4>
        Roles
      </h4>
      <p />
    </aside>
    <article
      className="col-sm-7 col-xs-12"
      role="main">
      <section>
        <h2>
          Challenge
        </h2>
        <p
          className="freetext">
          The My Sydney site and campaign provides a way for Sydney siders and visitors to find out what’s happening in the CBD in terms of transport changes. Gather requirements and provide a solution to allow people to interact with the maps on the My Sydney site. We had to acheive this while working with a separate digital agency who delivered the main website, campaign material and design direction.
        </p>
      </section>
      <section>
        <h2>
          Approach
        </h2>
        <p
          className="freetext">
          We were not sure exactly what the requirements were at the beginning of the project. Ascension worked with the client as well as other vendors to come to an agreement on scope together. As a result of the requirements, we realised that a bespoke CMS-based map editing tool had to be built. Even with changing requirements Ascension was able to deliver all products. We did this by staying light and agile in our approach to design and development. Due to Ascensions\' multi-disciplinary background and approach, we were able to recommend solutions encompassing UX guidance and development constraints with efficiency of work and confidence in knowledge site. We were required to understand and analyse the first two in order to decide what was most appropriate to fit into the new maps. The final solution required us to develop a CMS that included a map editing tool, an API for both us and the other vendor to interface with the data, and a workflow for publishing CMS changes. Although challenging, we were able to deliver holistic solutions by having front-end developers, back-end developers and the CMS users to all work together to reach a model together.
        </p>
      </section>
      <section>
        <h2>
          Outcomes and benefits
        </h2>
        <ul>
          <li>
            We gathered all requirements for maps including what was to be shown, what data was available.
          </li>
          <li>
            We recommended solutions encompassing UX guidance and development constraints with efficiency of work and confidence in knowledge.
          </li>
          <li>
            We developed a CMS that included a map editing tool, an API for both us and the other vendor to interface with the data, and a workflow for publishing CMS changes.
          </li>
        </ul>
      </section>
      <div
        className="casestudy__actions" />
    </article>
  </div>
</section>
`;

exports[`test View with required attributes and meta links 1`] = `
<section
  id="casestudy__view">
  <header
    className="row">
    <div
      className="col-xs-12">
      <h1>
        My Sydney
      </h1>
    </div>
    <div
      className="meta col-xs-12">
      <div
        className="row">
        <div
          className="col-xs-12 col-sm-7">
          <p>
            Transport for NSW
          </p>
        </div>
        <div
          className="col-xs-12 col-sm-5 actions">
          <a
            href="link/to/edit">
            Edit case study
          </a>
          <button
            className="button-secondary"
            onClick={[Function]}>
            Delete
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    className="row">
    <aside
      className="col-sm-3 col-xs-12">
      <h4>
        Timeframe
      </h4>
      <p>
        January 2016 — June 2016
      </p>
      <h4>
        Service
      </h4>
      <p />
      <h4>
        Roles
      </h4>
      <p />
    </aside>
    <article
      className="col-sm-7 col-xs-12"
      role="main">
      <section>
        <h2>
          Challenge
        </h2>
        <p
          className="freetext">
          The My Sydney site and campaign provides a way for Sydney siders and visitors to find out what’s happening in the CBD in terms of transport changes. Gather requirements and provide a solution to allow people to interact with the maps on the My Sydney site. We had to acheive this while working with a separate digital agency who delivered the main website, campaign material and design direction.
        </p>
      </section>
      <section>
        <h2>
          Approach
        </h2>
        <p
          className="freetext">
          We were not sure exactly what the requirements were at the beginning of the project. Ascension worked with the client as well as other vendors to come to an agreement on scope together. As a result of the requirements, we realised that a bespoke CMS-based map editing tool had to be built. Even with changing requirements Ascension was able to deliver all products. We did this by staying light and agile in our approach to design and development. Due to Ascensions\' multi-disciplinary background and approach, we were able to recommend solutions encompassing UX guidance and development constraints with efficiency of work and confidence in knowledge site. We were required to understand and analyse the first two in order to decide what was most appropriate to fit into the new maps. The final solution required us to develop a CMS that included a map editing tool, an API for both us and the other vendor to interface with the data, and a workflow for publishing CMS changes. Although challenging, we were able to deliver holistic solutions by having front-end developers, back-end developers and the CMS users to all work together to reach a model together.
        </p>
      </section>
      <section>
        <h2>
          Outcomes and benefits
        </h2>
        <ul>
          <li>
            We gathered all requirements for maps including what was to be shown, what data was available.
          </li>
          <li>
            We recommended solutions encompassing UX guidance and development constraints with efficiency of work and confidence in knowledge.
          </li>
          <li>
            We developed a CMS that included a map editing tool, an API for both us and the other vendor to interface with the data, and a workflow for publishing CMS changes.
          </li>
        </ul>
      </section>
      <div
        className="casestudy__actions" />
    </article>
  </div>
</section>
`;
